core/string/string_name.h

```diff
@@ -146,19 +148,25 @@ class StringName {
 	};
 
 	void operator=(const StringName &p_name);
-	StringName(const char *p_name);
+	StringName(const char *p_name, bool p_static = false);
 	StringName(const StringName &p_name);
-	StringName(const String &p_name);
-	StringName(const StaticCString &p_static_string);
+	StringName(const String &p_name, bool p_static = false);
+	StringName(const StaticCString &p_static_string, bool p_static = false);
 	StringName() {}
-	~StringName();
+	_FORCE_INLINE_ ~StringName() {
```

Comment:
Just curious, did you need to add `_FORCE_INLINE_` on the destructor for specific c++ behavior or it was for performance reason? Maybe add a short comment here could be useful for someone else that will check this in future.

Reply:
Its for performance, Variant has a similar one.

Reply:
Ok thanks!